repo.go

```diff
@@ -97,6 +99,13 @@ func NewRepoIndent(local LocalStore, prefix string, indent string,
 	if err != nil {
 		return nil, err
 	}
+	// Stage a new root if one isn't present
+	if _, err := r.root(); err != nil && errors.Is(err, errUninitializedRoot) {
+		if _, err = r.NewRoot(); err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
@asraa Could you please explain what this part of the code does and does it have anything to do with why `r.RootVersion()` (and other methods calling `r.root()`) doesn't get an error when it calls `r.root()`?